specs/heading_attrs.txt

```diff
@@ -0,0 +1,554 @@
+Examples and edge cases for attribute blocks for headings.
+
+# Basic usage
+
+Attribute blocks are attributes wrapped by curly braces (`{` and `}`) at the
+end of heading content.
+
+Strictly speaking, attribute blocks satisfies the conditions below:
+
+* Placed at the end of the heading, with optional following spaces.
+* Starts with `{` and ends with `}`.
+* Does not contain any other `{` and `}` except for opening and ending of the block.
+* Does not contain any `<`s (less-than operators), `>`s (greater-than operators),
+  `\`s (backslashes), or newline characters.
+
+Attributes are space-separated.
```

Comment:
Nit: ascii non-newline whitespace is okay, right?

Reply:
Ah... It was ambiguous.
I intended only ASCII whitespaces, as "whitespaces" in other part of the headings are also ASCII whitespaces.

For example, `"# H1 "` is converted to `"<h1>H1</h1>"` (U+0020 suffix is trimmed), but `"# H1\u{2000}"` is converted to `"<h1>H1\u{2000}</h1>"` (where U+2000 EN QUAD suffix is not trimmed, while its category is Space Separator, same as U+0020).

[Handling in other implementations](https://babelmark.github.io/?text=%23+H1+%7B+%23id%E2%80%8B%E2%80%80.class+%7D) seems various, but none of them does not accept U+2000 as a separator whitespace.

I think I should change the line to "Attributes are separated by ASCII whitespaces.".
What do you think?

Reply:
Yes, I intended "ASCII non-newline whitespace".
More strictly, "the characters considered as a whitespace by [`u8::is_ascii_whitespace`](https://doc.rust-lang.org/stable/std/primitive.u8.html#method.is_ascii_whitespace)".

It does **not** accept vertical tab (U+000B) as a whitespace, so I'll clarify that and add test cases.

Reply:
Vertical tab is very weird -- let's exclude it!